src/osa/workflow/stages.py

```diff
@@ -121,9 +122,19 @@ def _remove_dl1b_output(self, file_prefix):
 
     def _write_checkpoint(self):
         """Write the checkpoint in the history file."""
+        dl1_commands = [
+            cfg.get("lstchain", "r0_to_dl1"),
+            cfg.get("lstchain", "dl1ab"),
+            cfg.get("lstchain", "check_dl1")
+            ]
```

Comment:
might be better to create a dictionary matching the command to the prod id:

```python
command_to_prod_id = {
    cfg.get("lstchain", "r0_to_dl1"): options.prod_id,
    cfg.get("lstchain", "dl1ab"): options.dl1_prod_id,
    cfg.get("lstchain", "check_dl1"): options.dl1_prod_id,
    cfg.get("lstchain", "dl1_to_dl2"): options.dl2_prod_id
}
```
then:

```python
prod_id = command_to_prod_id.get(self.command)
```